src/assets/RustAsset.js

```diff
@@ -0,0 +1,93 @@
+const url = require('url');
+const path = require('path');
+const toml = require('toml');
+const {exec} = require('child-process-promise');
+
+const fs = require('../utils/fs');
+const JSAsset = require('./JSAsset');
+const localRequire = require('../utils/localRequire');
+
+const rustTarget = `wasm32-unknown-emscripten`;
+
+class RustAsset extends JSAsset {
+  collectDependencies() {
+    // Do nothing. Dependencies are collected by cargo :).
+  }
+
+  async parse(code) {
+    this.invalidateBundle();
+    this.invalidate();
+    this.dependencies.clear();
+    const {name, encoding, options} = this;
+    const release = process.env.NODE_ENV === 'production';
+    const cmd = `wargo build ${release ? ' --release' : ''}`;
```

Comment:
did you mean `cargo`?

Reply:
no, wargo is a library that setup your environment and makes easy to compile rust into wasm

Reply:
Does this PR verify that wargo is installed and accessible or fail violently?

Reply:
right now it's failing violently, 😄  but I was working on an api more friendly to use wargo directly instead of using wargo through child_process. 
However, I was thinking about to stop using wargo since wargo is using `wasm32-unknown-emscripten` and I think `wasm32-unknown-unknown` is a better option...